op-program/host/kvstore/disk.go

```diff
@@ -0,0 +1,64 @@
+package kvstore
+
+import (
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+	"path"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// read/write mode for user/group/other, not executable.
+const diskPermission = 0666
+
+// DiskKV is a disk-backed key-value store, every key-value pair is a hex-encoded .txt file, with the value as content.
+// DiskKV is safe for concurrent use; Puts may conflict, but write the exact same data anyway.
+type DiskKV struct {
+	path string
+}
+
+// NewDiskKV creates a DiskKV that puts/gets pre-images as files in the given directory path.
+// The path must exist, or subsequent Put/Get calls will error when it does not.
+func NewDiskKV(path string) *DiskKV {
+	return &DiskKV{path: path}
+}
+
+func (d *DiskKV) pathKey(k common.Hash) string {
+	return path.Join(d.path, k.String()+".txt")
+}
+
+func (d *DiskKV) Put(k common.Hash, v []byte) error {
+	// no O_EXCL, the pre-image may already exist. It's fine to overwrite it.
+	f, err := os.OpenFile(d.pathKey(k), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, diskPermission)
```

Comment:
Won't this truncate the file on opening and create a potential race condition where the file exists but has no content (where previously it may have existed with content)?

We'd need to be doing atomic writes by writing to a separate file then an atomic move into place (assuming the filesystem supports atomic moves) or add a lock like in the memory kv store.

Reply:
Easy solution is to use a RWMutex - I'd be the performance would be acceptable.

Reply:
Since we're not supposed to overwrite pre-images with different values anyway, I think I'll add `os_O_EXCL` instead. I'll add an already-exists error to the Put interface.

Reply:
Even with O_EXCL I think you can race between reading & writing.

Reply:
Added a rw lock and updated docs to cover the concurrent safety issue if multiple different instances with the same disk directory are used.